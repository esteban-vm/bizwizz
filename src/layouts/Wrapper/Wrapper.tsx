import type { FC, PropsWithChildren } from 'react'
import { Wave } from '@/components'

type WrapperProps = PropsWithChildren<{
  title: Capitalize<string>
  subtitle?: Capitalize<string>
  label: Lowercase<string>
  light?: boolean
}>

const Wrapper: FC<WrapperProps> = ({ title, subtitle, label, light, children }) => {
  const titleId = `${label}_title`

  return (
    <section aria-labelledby={titleId} className={`${light ? 'bg-white' : 'bg-black'} pb-32 text-center`} id={label}>
      <Wave is={light ? 'black' : 'white'} />
      <header className='font-bold tracking-widest'>
        <h2 id={titleId} className='mb-5 text-[2.4rem] uppercase text-primary'>
          {title}
        </h2>
        <h3 className='mb-10 text-[4rem] capitalize text-darkgrey'>{subtitle}</h3>
      </header>
      {children}
    </section>
  )
}

Wrapper.defaultProps = {
  subtitle: 'Lorem ipsum dolor sit amet',
}

export default Wrapper
