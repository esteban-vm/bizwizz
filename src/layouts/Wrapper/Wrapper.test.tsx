import { render, cleanup, within, screen } from '@/tests'
import Wrapper from './Wrapper'

describe('Wrapper:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(
      <Wrapper title='Testing title' subtitle='Testing subtitle' label='test' light>
        <article>A child component</article>
      </Wrapper>
    ))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const wrapperElem = screen.getByRole('region')
    expect(wrapperElem).toBeInTheDocument()
    expect(wrapperElem).toHaveAttribute('id', 'test')
    const titleElem = within(wrapperElem).getByRole('heading', { name: /testing title/i, level: 2 })
    expect(titleElem).toBeInTheDocument()
    expect(titleElem).toHaveAttribute('id', 'test_title')
    expect(within(wrapperElem).getByRole('heading', { name: /testing subtitle/i, level: 3 })).toBeInTheDocument()
  })
})
