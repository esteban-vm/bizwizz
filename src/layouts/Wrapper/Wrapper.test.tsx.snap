// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Wrapper: > should render correctly 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="test_title"
    class="bg-white pb-32 text-center"
    id="test"
  >
    <svg
      class="text-black h-40 w-full"
      preserveAspectRatio="none"
      role="presentation"
      viewBox="0 0 500 150"
    >
      <path
        d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
        fill="currentColor"
        stroke="none"
      />
    </svg>
    <header
      class="font-bold tracking-widest"
    >
      <h2
        class="mb-5 text-[2.4rem] uppercase text-primary"
        id="test_title"
      >
        Testing title
      </h2>
      <h3
        class="mb-10 text-[4rem] capitalize text-darkgrey"
      >
        Testing subtitle
      </h3>
    </header>
    <article>
      A child component
    </article>
  </section>
</DocumentFragment>
`;
